### ISSUE TITLE
Host contamination in final bins

### ISSUE THREAD

**Issue by SilasK** [metagenome-atlas/atlas#362]
Question from @botellaflotante

For some reason I have some host contamination within the MAGs. But maybe I could filter that with MAG purify on the bins and rerun genomes...
Do you think this way of doing it would affect the assembly process differently?

best

**Comment by SilasK** [comment]
I think the host decontamination step is never perfect, due to several reasons:
- The reference genome might not correspond exactly to the host in your case.
- Mapping precission (which you can change in the config file)

There is also a trade-off between removing enough and removing too much. 

What could be done?

- Magpurify could be an option but you had not ideal results didn't you #324
- We are working on a taxonomy prediction on contigs #356 
So this could be used to remove contigs from your host. Let me know if you want to work with this atlas dev version.
- Remove based on coverage. 
This is the idea that the coverage of the contamination contigs correlate with the removed reads.

**Comment by botellaflotante** [comment]
Yes, the real host might be slightly different to the reference host I have given ATLAS... I haven't tried MAGpurify but I think check-contam module could be used over the bins, although I would need to change the host in the database, which is hard coded..

**Comment by SilasK** [comment]
Ah, yes you used refineM. I tested MAGpurify some time ago also with ambivalent results as you have for refioneM

**Comment by SilasK** [comment]
Hey @botellaflotante I had some more thoughts on your problem.

You have very deep-sequenced soil microbiome, haven't you? You get some bins but they are contaminated by your host (plant) genome. 

So there are some modules I'm about to implement in atlas:

1. vamb (machine learning) binner #365 which should work well if you have a lot of data
2. Taxonomic annotation of contigs #356 Which would do what you want to do with MAGpurify.

You can try them out if you want but you would need to work with atlas in dev mode. 

Obviously, the first step is to get a good assembly, which is complicated in soil. I wonder if you could use bbnorm (installed with atlas) to get different fractions of reads depending on their coverage. E.g assemble first very high abundant genomes , then lower abundant 
https://jgi.doe.gov/data-and-tools/bbtools/bb-tools-user-guide/bbnorm-guide/

What do you think?
